flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionVertexCancelTest.java

```diff
@@ -360,8 +361,23 @@ public void testActionsWhileCancelling() {
 				ExecutionVertex vertex = new ExecutionVertex(ejv, 0, new IntermediateResult[0],
 						AkkaUtils.getDefaultTimeout());
 
-				Instance instance = getInstance(new ActorTaskManagerGateway(DummyActorGateway.INSTANCE));
-				SimpleSlot slot = instance.allocateSimpleSlot();
+				final TestingSlotOwner slotOwner = new TestingSlotOwner();
+				final TaskManagerLocation taskManagerLocation = new LocalTaskManagerLocation();
+				slotOwner.setReturnAllocatedSlotConsumer((LogicalSlot logicalSlot) -> {
+					try {
+						Method m = Slot.class.getDeclaredMethod("markReleased");
+						m.setAccessible(true);
+						m.invoke(logicalSlot);
+					}
+					catch (Exception e) {
+						throw new RuntimeException("Calling the method failed", e);
+					}
+				});
```

Comment:
Let's not use reflection to change internals. This is super brittle and will explode eventually.